File: 1843-number-of-rectangles-that-can-form-the-largest-square/number-of-rectangles-that-can-form-the-largest-square.js
/**
 * @param {number[][]} rectangles
 * @return {number}
 */
var countGoodRectangles = function(rectangles) {
    // #as you iterate through the array, find a lower value
    //     #append that lower value to the list
    //     #find the maximum value from the list
    //     #iterate through the list that contains the lower values and increase the counter
    //     #every time the element is equal to the maximum value
    const lowerValues = [];
    for(let i = 0; i < rectangles.length; i++){
        const len = rectangles[i][0];
        const wid =  rectangles[i][1];
        const lowerValue = Math.min(len, wid);
        lowerValues.push(lowerValue);
    }
    const maxValue = Math.max(...lowerValues);
    let counter = 0;
    lowerValues.forEach(value => {
        if(value === maxValue){
            counter++;
        }
    })
    return counter;
};

File: 1843-number-of-rectangles-that-can-form-the-largest-square/number-of-rectangles-that-can-form-the-largest-square.py
class Solution:
    def countGoodRectangles(self, rectangles: List[List[int]]) -> int:
        #as you iterate through the array, find a lower value
        #append that lower value to the list
        #find the maximum value from the list
        #iterate through the list that contains the lower values and increase the counter
        #every time the element is equal to the maximum value
        lower_values = []
        max_value_counter = 0
        for rectangle in rectangles:
            length = rectangle[0]
            width = rectangle[1]
            min_value = min(length, width)
            lower_values.append(min_value)
        max_value = max(lower_values)
        for value in lower_values:
            if value == max_value:
                max_value_counter += 1
        return max_value_counter
